lib/Runtime/Base/FunctionBody.cpp

```diff
@@ -8830,28 +8830,6 @@ namespace Js
     {
         if (this->GetState() != CleanedUp)
         {
-            // Unregister xdataInfo before OnCleanup() which may release xdataInfo->address
```

Comment:
why are you removing this block?

Reply:
there's a duplication in OnCleanup...

Reply:
I don't think they cover the same cases, here is `this->GetState() != CleanedUp`, there is `GetState() == CodeGenDone`. Maybe we should combine into a single block, but I think we are now missing some cases.

Reply:
yeah, I'm fixing this part, so is it possible that it's not CodeGenDone but we have xdata allocated?
currently I'm adding an assertion, if yes, I'll change to just delete the function table in UI thread in that case.

Reply:
It is possible. In particular it is likely in case of `CodeGenRecorded`. XData registration happens around this point in time. Only once you call the new JIT entrypoint for the first time will we do final setup and change the state to `CodeGenDone`.